### PR TITLE
Update dependency libs  & newer mysql and k8s

### PR DESCRIPTION
**Pull request by takara9**
Notable change:
Due to an update of the `github.com/aws/aws-sdk-go-v2` library, users must specify the region of the bucket in their `BackupPolicy` if the `bucketConfig.backendType` is `s3` or not specified. If the region is not specified, the backup/restore job will crash.
ref: https://github.com/aws/aws-sdk-go-v2/issues/2502

Dependencies:
- Bump up controller-runtime 0.15.0
- Bump golang version 1.22.3

Additional Support :
- Support for MySQL version 8.0.36 and 8.0.37
- Support for Kubernetes version 1.28,1.29

### CODE REVIEW

File: .github/actions/dbtest/action.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - uses: actions/setup-go@v4
+    - uses: actions/setup-go@v5
```
Comment (masa213f) [inline .github/actions/dbtest/action.yaml:12]: Please check all actions other than `setup-go`, too.
Some actions are not updated even though new ones are being released.

File: kustomization.yaml [hunk 2]
```diff
@@ -3,4 +3,4 @@ resources:
 
 images:
 - name: ghcr.io/cybozu-go/moco
-  newTag: 0.20.2
```
Comment (masa213f) [inline kustomization.yaml:6]: We should not update here until MOCO v0.20.3 is released.

**Review by masa213f — CHANGES_REQUESTED** [review]

File: .github/workflows/build-mysql-container.yaml [hunk 3]
```diff
@@ -16,7 +16,7 @@ on:
 
 jobs:
   filter:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
[the remark this answers is not in the document]
Comment (takara9) [reply .github/workflows/build-mysql-container.yaml:19]: I find ubuntu-20.04 in source repository and did update for all findings.

File: .github/actions/dbtest/action.yaml (hunk 1 above)
Comment (takara9) [reply .github/actions/dbtest/action.yaml:12]: I find Actions in whole repository and if find it, changed it.

File: kustomization.yaml (hunk 2 above)
Comment (takara9) [reply kustomization.yaml]: I revert it to 0.20.2

File: pkg/bucket/s3_test.go [hunk 4]
```diff
@@ -45,8 +45,7 @@ var _ = Describe("S3Bucket", func() {
 				AccessKeyID:     "minioadmin",
 				SecretAccessKey: "minioadmin",
 				Source:          "minio default credentials",
-			},
-		}))
+			}}), config.WithRegion("us-west-2"))
```
Comment (masa213f) [inline pkg/bucket/s3_test.go:48]: Plase use [`WithRegion()`](https://github.com/cybozu-go/moco/blob/v0.20.2/pkg/bucket/s3.go#L30-L35).
And it's a trivial thing but the default region of MinIO seems to be `us-east-1`.

```suggestion
		cfg, err := config.LoadDefaultConfig(ctx, config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
			Value: aws.Credentials{
				AccessKeyID:     "minioadmin",
				SecretAccessKey: "minioadmin",
				Source:          "minio default credentials",
			}}))
		Expect(err).NotTo(HaveOccurred())
		client := s3.NewFromConfig(cfg,
			WithEndpointURL("http://localhost:9000"),
			WithRegion("us-east-1"), // Please specify here.
			WithPathStyle(),
		)
```

**Review by masa213f — CHANGES_REQUESTED** [review]

File: pkg/bucket/s3.go [hunk 5]
```diff
@@ -62,7 +62,7 @@ type s3Bucket struct {
 
 // NewS3Bucket creates a Bucket that manage object in S3.
 func NewS3Bucket(name string, optFns ...func(*s3.Options)) (Bucket, error) {
-	cfg, err := config.LoadDefaultConfig(context.Background())
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion("us-west-2"))
```
[the remark this answers is not in the document]
Comment (takara9) [reply pkg/bucket/s3.go:65]: @masa213f 
thank you for your suggetion.
I will revert it.

File: pkg/bucket/s3_test.go (hunk 4 above)
Comment (takara9) [reply pkg/bucket/s3_test.go:48]: @masa213f 
Thank you for your code that I should add.
I will change the code arroding to your recomendation.

Comment (masa213f) [reply pkg/bucket/s3_test.go]: Please reflect this comment.

File: docs/usage.md [hunk 6]
```diff
@@ -428,7 +429,8 @@ spec:
       value: minioadmin
     - name: AWS_SECRET_ACCESS_KEY
       value: minioadmin
-
+    - name: AWS_REGION
```
Comment (masa213f) [inline docs/usage.md:432]: Thanks for the change. Nice!

But, basically, I would like to guide users to use the `bucketConfig.region`.
Could you change this example to use `bucketConfig.region`.

File: docs/usage.md [hunk 7]
```diff
@@ -428,7 +429,8 @@ spec:
...
+      value: us-east-1
     # bucketName is required.  Other fields are optional.
```
Comment (masa213f) [inline docs/usage.md:434]: Please update this comment. When the `backendType` is `s3`(default). `region` is required.
```
    # bucketName is required.  Other fields are optional.
```

**Review by masa213f — CHANGES_REQUESTED** [review]

File: pkg/bucket/s3_test.go (hunk 4 above)
Comment (takara9) [reply pkg/bucket/s3_test.go]: Sorry, I don't undestand your this reply.
I changed region us-test-1,
 Is it not enough ?

Comment (masa213f) [reply pkg/bucket/s3_test.go]: Please read suggested code.

Comment (takara9) [reply pkg/bucket/s3_test.go]: @masa213f 
I changed suggested code.
Is it right ?

File: docs/usage.md (hunk 7 above)
Comment (takara9) [reply docs/usage.md:434]: @masa213f 
I don't well understand for backup of MOCO. 
but, I tried to make explanation for this s3 case config.
please review for is it right.

File: pkg/bucket/s3_test.go (hunk 4 above)
Comment (takara9) [reply pkg/bucket/s3_test.go]: @masa213f 
I updated for this

File: docs/usage.md (hunk 6 above)
Comment (takara9) [reply docs/usage.md]: I updated is for this change request.

File: docs/usage.md (hunk 7 above)
Comment (takara9) [reply docs/usage.md]: @masa213f 
I updated for your this request.

**Review by masa213f — CHANGES_REQUESTED** [review]

File: docs/usage.md [hunk 8]
```diff
@@ -428,13 +429,13 @@ spec:
...
-
-    # bucketName is required.  Other fields are optional.
+    # bucketName is mandatory, other key can be optional.
     bucketConfig:
       bucketName: moco
       endpointURL: http://minio.default.svc:9000
       usePathStyle: true
-
+      # backendType: gcs   # s3 is default value, If using s3, this line can omit.
+      # region: AWS_REGION # When backendType is s3, must specify AWS region. This line can be omittable in other case.
```
[the remark this answers is not in the document]
Comment (takara9) [reply docs/usage.md:438]: I re-write this accoding to your recomendation.

File: docs/usage.md [hunk 9]
```diff
@@ -472,6 +473,8 @@ spec:
       bucketName: "$(BUCKET_NAME)"
       endpointURL: http://minio.default.svc:9000
       usePathStyle: true
+      # backendType: gcs   # If using s3, this line can omit.
+      # region: AWS_REGION # When backendType is s3, must specify AWS region.
```
[the remark this answers is not in the document]
Comment (takara9) [reply docs/usage.md:477]: I will delete redundant lines.

File: docs/usage.md [hunk 10]
```diff
@@ -575,6 +578,8 @@ spec:
         bucketName: moco
         endpointURL: http://minio.default.svc:9000
         usePathStyle: true
+        # backendType: gcs   # If using s3, this line can omit.
+        # region: AWS_REGION # When backendType is s3, must specify AWS region.
```
[the remark this answers is not in the document]
Comment (takara9) [reply docs/usage.md:582]: I deleted redundant line and add "region;..."

**Review by masa213f — APPROVED** [review]
LGTM

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**